spec.md

```diff
@@ -2734,6 +2780,270 @@ message NodeExpandVolumeResponse {
+
+```protobuf
+message CreateVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // The suggested name for the group snapshot. This field is REQUIRED
+  // for idempotency.
+  // Any Unicode string that conforms to the length limit is allowed
+  // except those containing the following banned characters:
+  // U+0000-U+0008, U+000B, U+000C, U+000E-U+001F, U+007F-U+009F.
+  // (These are control characters other than commonly used whitespace.)
+  string name = 1;
+
+  // volume ids of the source volumes to be snapshotted together.
+  // This field is REQUIRED.
+  repeated string source_volume_ids = 2;
+
+  // Secrets required by plugin to complete
+  // ControllerCreateVolumeGroupSnapshot request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  // The secrets provided in this field SHOULD be the same as
+  // the secrets provided in ControllerDeleteVolumeGroupSnapshot
+  // and ControllerGetVolumeGroupSnapshot requests for the same
+  // group snapshot unless if secrets are rotated after the
+  // group snapshot is created.
+  map<string, string> secrets = 3 [(csi_secret) = true];
+
+  // Volume secrets required by plugin to complete volume group
+  // snapshot creation request. This field is needed in case the
+  // volume level secrets are different from the above secrets
+  // for the group snapshot.
+  // This field is OPTIONAL.
+  repeated VolumeSecret volume_secrets = 4;
+
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  // This field is OPTIONAL. The Plugin is responsible for parsing and
+  // validating these parameters. COs will treat these as opaque.
+  map<string, string> parameters = 5;
+}
+
+message VolumeSecret {
+  // ID of the volume whose secrets are provided.
+  // This field is REQUIRED.
+  string volume_id = 1;
+
+  // Secrets required by plugin for a volume operation.
+  // This field is REQUIRED. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 2 [(.csi.v1.csi_secret) = true];
+}
+
+message CreateVolumeGroupSnapshotResponse {
+  option (alpha_message) = true;
+
+  // Contains all attributes of the newly created group snapshot.
+  // This field is REQUIRED.
+  VolumeGroupSnapshot group_snapshot = 1;
+}
+
+message VolumeGroupSnapshot {
+  option (alpha_message) = true;
+
+  // The identifier for this group snapshot, generated by the plugin.
+  // This field MUST contain enough information to uniquely identify
+  // this specific snapshot vs all other group snapshots supported by
+  // this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this group snapshot.
+  // The SP is NOT responsible for global uniqueness of
+  // group_snapshot_id across multiple SPs.
+  // This field is REQUIRED.
+  string group_snapshot_id = 1;
+
+  // A list of snapshots created.
```

Comment:
suggestion: "A list of snapshots belonging to this group."

Reply:
Done